jerry-core/api/jerry-debugger.c

```diff
@@ -116,3 +116,81 @@ jerry_debugger_cleanup (void)
   }
 #endif /* JERRY_DEBUGGER */
 } /* jerry_debugger_cleanup */
+
+/**
+ * Sets whether the engine should wait and run a source.
+ *
+ * @return true - if the client source has been processed
+ *         false - otherwise
+ */
+bool
+jerry_debugger_wait_and_run_client_source (void)
+{
+#ifdef JERRY_DEBUGGER
+  if ((JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED)
+      && !(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_BREAKPOINT_MODE))
+  {
+    JERRY_CONTEXT (debugger_flags) = (uint8_t) (JERRY_CONTEXT (debugger_flags) | JERRY_DEBUGGER_CLIENT_SOURCE_MODE);
+    jerry_debugger_uint8_data_t *client_source_data_p = NULL;
+
+    while (true)
+    {
+      if (jerry_debugger_receive (&client_source_data_p))
+      {
+        if (!(JERRY_CONTEXT (debugger_flags) & JERRY_DEBUGGER_CONNECTED))
+        {
+          jmem_heap_free_block (client_source_data_p,
```

Comment:
This is incorrect. `client_source_data_p` can be NULL. What about simply break here and let the `if` handle the free below.